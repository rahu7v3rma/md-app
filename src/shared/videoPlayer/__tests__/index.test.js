import React from 'react';
import WebView from 'react-native-webview';
import { act, create } from 'react-test-renderer';

import VideoPlayer from '@/shared/videoPlayer';

test('snapshot', () => {
    expect(create(<VideoPlayer />).toJSON()).toMatchSnapshot();
});

test('renders the video with the correct URL', () => {
    const mockVideoId = 'mockVideoId';
    const params =
        'api=1&autoplay=0&byline=0&portrait=0&title=0&playsinline=1&autopause=true';
    expect(
        create(<VideoPlayer videoId={mockVideoId} />).root.findByType(WebView)
            .props.source
    ).toMatchObject({
        uri: `https://player.vimeo.com/video/${mockVideoId}?${params}`
    });
});

test('applies custom styles to the video player', () => {
    expect(
        create(
            <VideoPlayer videoPlayerstyle={{ borderColor: 'blue' }} />
        ).root.findByType(WebView).props.style
    ).toEqual([
        {
            width: '100%',
            height: '100%',
            resizeMode: 'cover'
        },
        { borderColor: 'blue' }
    ]);
});

test('calls the onErr callback on HTTP error', async () => {
    const mockOnErr = jest.fn();
    const videoPlayerComponent = create(
        <VideoPlayer onErr={mockOnErr} />
    ).root.findByType(WebView);
    await act(() => videoPlayerComponent.props.onHttpError());
    expect(mockOnErr).toBeCalled();
});
