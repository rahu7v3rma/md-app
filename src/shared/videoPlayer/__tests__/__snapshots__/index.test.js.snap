// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    [
      {
        "borderRadius": 8,
        "overflow": "hidden",
        "width": "100%",
      },
      undefined,
      {
        "height": undefined,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      allowsFullscreenVideo={true}
      allowsInlineMediaPlayback={true}
      cacheEnabled={true}
      contentMode="mobile"
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      scrollEnabled={false}
      source={
        {
          "uri": "https://player.vimeo.com/video/undefined?api=1&autoplay=0&byline=0&portrait=0&title=0&playsinline=1&autopause=true",
        }
      }
      style={
        [
          {
            "flex": 1,
            "overflow": "hidden",
          },
          {
            "backgroundColor": "#ffffff",
          },
          [
            {
              "height": "100%",
              "resizeMode": "cover",
              "width": "100%",
            },
            undefined,
          ],
        ]
      }
      textInteractionEnabled={true}
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
